src/manifest/manifestparser.js

```diff
@@ -26,26 +29,34 @@ function parseMPD (manifest, dateWithOffset) {
         correction: timeCorrection
       }
     } else {
-      return { error: 'Error parsing DASH manifest attributes' }
+      throw new Error('manifest-dash-attributes-parse-error')
     }
   } catch (e) {
-    return { error: 'Error parsing DASH manifest' }
+    const error = new Error(e.message || 'manifest-dash-parse-error')
```

Comment:
Will we not get a different message here if it actually fails on the try/catch? Do we not want it to always be either 'manifest-dash-parse-error' or 'manifest-dash-attributes-parse-error'

Also, we only ever report the single code and we have multiple paths of errors here? do we want to report this extra information?

Reply:
Yeah you end up with the different message but the same code, both of which are reported to the plugin. It is a bit annoying having to rely on the error throwing upwards.